MyHome.DataClasses/Expense.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+
+namespace MyHome.DataClasses
+{
+    public class Expense
+    {
+        public Expense(decimal amount, DateTime date, ExpenseCategory expenseCategory,
+            PaymentMethod paymentMethod, string comment, int id = 0)
+        {
+            Amount = amount;
+            Category = expenseCategory;
+            Comment = comment;
+            Date = date;
+            Id = id;
+            Method = paymentMethod;
+        }
+
+        /// <summary>
+        ///     The amount of the expense
+        /// </summary>
+        public decimal Amount { get; set; }
+
+        /// <summary>
+        ///     The date of the expense
+        /// </summary>
+        public DateTime Date { get; set; }
+
+
+        public int CategoryId { get; set; }
+
+        public int PaymentMethodId { get; set; }
+
+        /// <summary>
+        ///     Category of the expense
+        /// </summary>
+        public ExpenseCategory Category { get; set; }
+
+        /// <summary>
+        ///     How the expense was payed
+        /// </summary>
+        public PaymentMethod Method { get; set; }
+
+        /// <summary>
+        ///     Additional info about the expense
+        /// </summary>
+        public string Comment { get; set; }
+
+        /// <summary>
+        ///     ID number of the expense in the data table
+        /// </summary>
+        public int Id { get; set; }
+
+        public override bool Equals(object obj)
+        {
+            var expenseComparing = (Expense) obj;
+
+            return ((Amount == expenseComparing.Amount) &&
+                    (Category.Equals(expenseComparing.Category)) &&
+                    (Comment == expenseComparing.Comment) &&
+                    (Date == expenseComparing.Date) &&
+                    (Id == expenseComparing.Id) &&
+                    (Method.Equals(expenseComparing.Method)));
+        }
+
+        // TODO - QC - override get hash code as well.
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
```

Comment:
I will apply the second response in this post as the overhead mentioned in the comment is not a problem.

http://stackoverflow.com/questions/263400/what-is-the-best-algorithm-for-an-overridden-system-object-gethashcode